### ISSUE TITLE
Boolean parameter validation with suffix expansion edge case

### ISSUE THREAD

**Issue by hdoupe** [open-source-economics/Tax-Calculator#1980]
I received an unexpected error when running a JSON reform that specifies a switch parameter by its suffix.

```
import taxcalc


json_reform = """
{
    "policy": {
        "_ID_BenefitSurtax_Switch_medical": {"2018": [true]}
    }
}
"""

print('taxcalc version', taxcalc.__version__)

policy_dict = taxcalc.Calculator.read_json_param_objects(
    json_reform,
    None
)

pol = taxcalc.Policy()
pol.implement_reform(policy_dict["policy"])

print('Warnings:', pol.parameter_warnings)
print('Errors:', pol.parameter_errors)
```

output:

```
(taxcalc-dev) HDoupe-MacBook-Pro:taxcalc henrydoupe$ python ../switch_example.py 
taxcalc version 0.19.0-2-g19098dfc-dirty
Traceback (most recent call last):
  File "../switch_example.py", line 20, in <module>
    pol.implement_reform(policy_dict["policy"])
  File "/Users/henrydoupe/Documents/Tax-Calculator/taxcalc/policy.py", line 211, in implement_reform
    raise ValueError(self.parameter_errors)
ValueError: ERROR: 2018 _ID_BenefitSurtax_Switch_1 value 1 is not boolean
ERROR: 2018 _ID_BenefitSurtax_Switch_2 value 1 is not boolean
ERROR: 2018 _ID_BenefitSurtax_Switch_3 value 1 is not boolean
ERROR: 2018 _ID_BenefitSurtax_Switch_4 value 1 is not boolean
ERROR: 2018 _ID_BenefitSurtax_Switch_5 value 1 is not boolean
ERROR: 2018 _ID_BenefitSurtax_Switch_6 value 1 is not boolean

```

There should not be any errors for this reform. When the array is filled in, the type is changed to `np.int8` which causes a type error.

**Comment by hdoupe** [comment]
One solution could be to extend the array as boolean type instead of integer type if the type is supposed to be boolean. Then, `_expand_array` would look like:

```
    @staticmethod
    def _expand_array(x, x_dtype_int, x_dtype_bool, inflate, inflation_rates, num_years):
        """
        Private method called only within this abstract base class.
        Dispatch to either _expand_1D or _expand_2D given dimension of x.

        Parameters
        ----------
        x : value to expand
            x must be either a scalar list or a 1D numpy array, or
            x must be either a list of scalar lists or a 2D numpy array

        x_dtype_int : integer type
            True implies dtype=np.int8; False implies dtype=np.float64 
                    or np.bool_

        x_dtype_bool : boolean type
            True implies dtype=np.bool_; False implies dtype=np.float64 
                    or np.int8

        inflate: boolean
            As we expand, inflate values if this is True, otherwise, just copy

        inflation_rates: list of inflation rates
            Annual decimal inflation rates

        num_years: int
            Number of budget years to expand

        Returns
        -------
        expanded numpy array with specified dtype
        """
        assert not (x_dtype_int and x_dtype_bool)
        if not isinstance(x, list) and not isinstance(x, np.ndarray):
            msg = '_expand_array expects x to be a list or numpy array'
            raise ValueError(msg)
        if isinstance(x, list):
            if x_dtype_int:
                x = np.array(x, np.int8)
            elif x_dtype_bool:
                x = np.array(x, np.bool_)
            else:
                x = np.array(x, np.float64)
        if len(x.shape) == 1:
            return ParametersBase._expand_1D(x, inflate, inflation_rates,
                                             num_years)
        elif len(x.shape) == 2:
            return ParametersBase._expand_2D(x, inflate, inflation_rates,
                                             num_years)
        else:
            raise ValueError('_expand_array expects a 1D or 2D array')

```

**Comment by martinholmer** [comment]
@hdoupe,  Thanks for the bug report and suggested fix.

**Comment by hdoupe** [comment]
No problem. Thanks for the quick response @martinholmer

**Comment by martinholmer** [comment]
Issue #1980 has been resolved by pull request #1982.
